### DISCUSSION TITLE
`npm run dev` throws vite error if `.vite` cache folder is not empty

### DISCUSSION THREAD

**Discussion by b2m9** [sveltejs/kit#3418] [Q&A] [answered]
Hi there :wave: 

I updated a project from svelte kit `next.52` to `next.232` and I'm running into problems with `vite` loading PostCSS/Tailwind. I can run `npm run dev` and `npm run build`. However, I cannot run `npm run dev` a _second time_ once vite's cache in `node_modules/.vite` is not empty:

```sh
[vite] Error when evaluating SSR module /src/routes/__layout.svelte:
Error: Cannot find module '/home/.../postcss.config.cjs' imported from '/home/.../node_modules/vite/dist/node/chunks/dep-f5552faa.js'
```

Once I clear the `.vite` folder, it works again. According to the vite docs you can disable the cache by using `--force` but svelte kit doesn't seem to expose this flag. So my solution is just to nuke the folder via `rm -r node_modules/.vite && svelte-kit dev --host` for `npm run dev`

Has anyone encountered this problem? How can I convince vite not to import my PostCSS config?

Thankful for any hints :pray:

**Comment by benmccann** [comment]
I have no issues running Tailwind with `next.232`. I set it up with https://github.com/svelte-add/tailwindcss and recommend mirroring your project setup to what that would do

**Reply by b2m9** [reply]
I've checked that and even so vite says it's an error with CSS, I don't think that's true. I started with a new project and eventually ran into the same issue again.

The error only happens when I have `localhost:3000` open in the browser and restart the dev server. Everything is fine, if I close the browser window and restart the dev server. I have no idea what to think of that.

The app is using `adapter-static`. I can't wrap my head around when exactly the dev server wants to import the PostCSS config. Gave up on it for the moment since it works just fine when there's no open window to the dev server when restarting it. It's such an odd behavior :shrug:

**Reply by akiarostami** [reply]
I am having a similar issue. I did a lot of install/uninstall, and worked up until v249. After I updated to v250, I started getting:

```
[vite] Internal server error: Cannot find module '[...]/package.json' imported from '[...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js'
  Plugin: vite:css
  File: [...]/src/app.css
```
  
@benmccann any idea what has happened, and how I can fix it?

**Reply by akiarostami** [reply]
Actually I take that back. Sometimes I have that problem with v249 as well, and sometimes I don't. It's been a very frustrating experience, and I can't even find a clear pattern why this happens.

Here's the full error message (paths removed):

```
[plugin:vite:css] Cannot find module '[...]/package.json' imported from '[...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js'
[...]/src/app.css
    at viteResolve ([...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js:60126:25)
    at Function._resolveFilename ([...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js:60140:28)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.search ([...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js:12705:33)
    at async resolvePostcssConfig ([...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js:20506:22)
    at async compileCSS ([...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js:20329:27)
    at async TransformContext.transform ([...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js:20015:50)
    at async Object.transform ([...]/node_modules/vite/dist/node/chunks/dep-f5552faa.js:36985:30

```

**Reply by benmccann** [reply]
This really is not enough info to diagnose and reproduce the issue. I'd recommend you file an issue providing all the requested information such as a repository that reproduces the issue

**Reply by akiarostami** [reply]
I understand this is not enough, but it's really all I have. I was hoping that this would be a known issue. I'll play with it more and see if I can reproduce it in a separate repository. Thank you!

**Reply by akiarostami** [reply]
@benmccann the issue looks very similar to [this one](https://github.com/sveltejs/vite-plugin-svelte/issues/263), which supposed to be fixed with `@sveltejs/vite-plugin-svelte` 1.0.0-next.35. Could it be that this issue is still not resolved?

**Reply by benmccann** [reply]
No, that issue is resolved. You are having a different issue

**Reply by benmccann** [reply]
I actually just ran into this problem on another project. It seems like upgrading to Vite 2.8 beta 5 fixed it. The final release of 2.8 will come out soon (Monday perhaps), so hopefully that will address your issue

**Reply by akiarostami** [reply]
Thank you @benmccann! Upgrading Vite solved my problem as well. I really appreaciate it!

**Reply by akiarostami** [reply]
@benmccann after I updated Vite, everything worked for half an hour. Then, without any change in the configurations or packages, it stopped working again with the same problem. Very frustrating...

**Reply by akiarostami** [reply]
@benmccann I'm still having this strange issue. As I mentioned, I installed the last Vite (2.8-beta.5, and today to beta 6) and things worked for half hour, before I started seeing this issue again (without changing anything in the packages or configurations).
Did updating to the latest Vite completely solve your issue?

**Reply by bluwy** [reply]
@akiarostami Are you on windows? There was a followup fix for that in Vite 2.8.0-beta.6

**Reply by akiarostami** [reply]
@bluwy no I’m on Mac. And I installed beta.26 earlier today as well, but still have that issue.

**Reply by bluwy** [reply]
That's really strange. Can you show the error messages with all the paths? Or better with a minimal repro repo/stackblitz. You can also debug this locally with Vite. The particular issue happens [here](https://github.com/vitejs/vite/blob/a0126441a556b4991ac14cf037820194ab9e17b9/packages/vite/src/node/ssr/ssrModuleLoader.ts#L234-L260). From the logic there, it has certain hueristics to avoid a "vite resolve" and resort to the default `"nodejs resolve" instead. Usually "vite resolve" is causing the error you see now.

**Reply by benmccann** [reply]
I can't remember which project I ran into this on. Maybe it was luck that I didn't encounter it again after upgrading Vite

**Reply by b2m9** [reply]
I can confirm that this behavior is gone in my project since SvelteKit updated its vite dependency in 262.

**Comment by hyunbinseo** [comment]
I did not encounter such issue using `tailwindcss`, `postcss`, `autoprefixer`.

I followed the official documentation, [Get started with Tailwind CSS](https://tailwindcss.com/docs/installation/using-postcss).

Could there be a repository for reproduction?

**Comment by hyunbinseo** [comment]
@b2m9 @akiarostami Providing a working repository instead.

~~https://github.com/hyunbinseo/sveltekit-tailwindcss~~

Run `npm run dev` and then route to `/tailwind`.

It should look like this [Hero](https://tailwinduikit.com/components/marketing/page_section/hero/centre_aligned_with_image) UI from Tailwind UI Kit.

**Reply by akiarostami** [reply]
Thank you @hyunbinseo, but I have a lot of code and copy/pasting it into a new repository does not make sense. Everything was working fine until I updated sveltekit and postcss. I'm hoping that someone would help me figure out what the problem is.

**Comment by b2m9** [comment, marked as answer]
This seems to be have been an issue with `vite`. It has been resolved in https://github.com/sveltejs/kit/pull/3791. I do not experience this issue anymore in version `1.0.0-next.262`.
